include/nanobind/nb_types.h

```diff
@@ -819,24 +819,29 @@ struct fast_iterator {
 
 class dict_iterator {
 public:
+    NB_NONCOPYABLE(dict_iterator)
```

Comment:
@wjakob can you please suggest what would be the best way to fix a thirdparty code using dict_iterator copy and now failing to compile due to this change?
Here is the code:
https://github.com/openxla/xla/blob/182111eb601b6491b37625df51f488905d217ae2/xla/python/weakref_lru_cache.cc#L58-L60
Thanks!

Reply:
This looks complicated (weak references, LRU caches, dictionary iterators, ...) Could I ask you to explain in a bit more what the use case is, and why dict iterators need to be copied and hashed?

Maybe that code can be rewritten without needing a dictionary iterator. Or perhaps it can call `.items()` and then run a normal iterator over that.

The change made in the commit is difficult to roll back or avoid since it needed to make `PyDict_Next` safe in free-threaded builds.